components/global/global_header.tsx

```diff
@@ -34,6 +34,12 @@ const GlobalHeaderContainer = styled.header`
     border-bottom: solid 1px rgba(var(--center-channel-color-rgb), 0.08);
     color: rgba(var(--global-header-text-rgb), 0.64);
     padding: 0 16px 0 12px;
+    z-index: 2;
+
+    + * {
```

Comment:
Here you are altering the `position` and the `z-index` of all the plugins and the custom routes from the outside an using a specific component styles to do it. It sounds wrong to me.

Reply:
It only affects the single sibling element immediately following the GH when the GH is enabled.

See here: https://community-daily.mattermost.com/core/pl/9pkmuzj85trrdqkqtcqdwqm9ne

It's equivalent to changing the GH z-index to a really high number to move it on top of any other possible element with a currently higher z-index, only this is much more controlled and doesn't require sifting through the dozens of defined z-index values to try find the perfect value. 

That said, I'm perfectly fine if individual products would prefer to handle layering themselves as is already done for channels.

Reply:
Still, I don't see it at Global Header level, having a component there that affects the siblings is something that looks wrong to me. For me, makes more sense to do it from the container of `GlobalHeader`, that can be a style that says `.GlobalHeader + * {`, but at the parent node, because that one has responsibility over all the implicated nodes. Not a big deal, but affecting sibling components from inside the component is a pattern that I don't like.

Because this is not an stopper in any case, I going to approve the PR.

Reply:
Ah yes, that is a very fair point and I absolutely agree ... I actually had the original channels-only implementation being applied by the parent, but got caught up in the weeds and forgot about that. 😛

One sec, I'll quickly modify!